tests/metrics/functional/statistical/test_wasserstein.py

```diff
@@ -0,0 +1,255 @@
+
+        if x.ndim == 1:
+            scipy_result = np.stack(
+                [
+                    sp_wasserstein(sp_x, sp_y, sp_x_w, sp_y_w)
+                    for sp_x, sp_y, sp_x_w, sp_y_w in zip(
+                        [x_np], [y_np], [x_weights_np], [y_weights_np]
+                    )
+                ]
+            )
+        else:
+            scipy_result = np.stack(
+                [
+                    sp_wasserstein(sp_x, sp_y, sp_x_w, sp_y_w)
+                    for sp_x, sp_y, sp_x_w, sp_y_w in zip(
+                        x_np, y_np, x_weights_np, y_weights_np
+                    )
+                ]
+            )
+
+        return torch.tensor(scipy_result, device=device).to(torch.float)
+
+    def _test_wasserstein1d_with_input(
+        self,
+        compute_result: torch.Tensor,
+        x: torch.Tensor,
+        y: torch.Tensor,
+        x_weights: Optional[torch.Tensor] = None,
+        y_weights: Optional[torch.Tensor] = None,
+    ) -> None:
+        if x.ndim == 1:
+            my_compute_result = wasserstein_1d(x, y, x_weights, y_weights)
+            torch.testing.assert_close(
+                my_compute_result,
+                compute_result,
+                equal_nan=True,
+                atol=1e-8,
+                rtol=1e-5,
+            )
+
+            # Also test for cuda
+            if torch.cuda.is_available():
+                compute_result_cuda = tuple(c.to("cuda") for c in compute_result)
+                my_compute_result_cuda = tuple(c.to("cuda") for c in my_compute_result)
+
+                torch.testing.assert_close(
+                    my_compute_result_cuda,
+                    compute_result_cuda,
+                    equal_nan=True,
+                    atol=1e-8,
+                    rtol=1e-5,
+                )
+        else:
+            my_compute_result = torch.Tensor(
+                [
+                    wasserstein_1d(x, y, x_weights, y_weights)
+                    for x, y, x_weights, y_weights in zip(x, y, x_weights, y_weights)
+                ]
+            ).to(x.device)
+            torch.testing.assert_close(
+                my_compute_result,
+                compute_result,
+                equal_nan=True,
+                atol=1e-8,
+                rtol=1e-5,
+            )
+
+            # Also test for cuda
+            if torch.cuda.is_available():
+                compute_result_cuda = tuple(c.to("cuda") for c in compute_result)
+                my_compute_result_cuda = tuple(c.to("cuda") for c in my_compute_result)
+
+                torch.testing.assert_close(
+                    my_compute_result_cuda,
+                    compute_result_cuda,
+                    equal_nan=True,
+                    atol=1e-8,
+                    rtol=1e-5,
+                )
+
+    def test_wasserstein1d_distribution_values_only(self) -> None:
+        x = torch.Tensor([5, -5, -7, 9, -3])
+        y = torch.Tensor([9, -7, 5, -4, -2])
```

Comment:
here and everywhere else in the tests: use `torch.tensor` for the constructor
```suggestion
        x = torch.tensor([5, -5, -7, 9, -3])
        y = torch.tensor([9, -7, 5, -4, -2])
```

Reply:
These changes have been implemented. Please help me check them, thanks!